### PR TITLE
[develop] Update documentation for Rocoto XML changes

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 

Updates the User's Guide sections on configuring the workflow, and added a section on Defining a Workflow.

I have not updated the FAQ or Tutorial pages. People using older versions may still need that information, so I'll leave it to other folks to handle that transition.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DOCUMENTATION:

This is only documentation.

## ISSUE: 

n/a

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [x] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 

@gspetro-NOAA

### CODE REVIEW

File: docs/UsersGuide/source/RunSRW.rst [hunk 1]
```diff
@@ -798,34 +687,49 @@ To use METplus verification, the path to the MET and METplus directories must be
...
+.. code-block:: console
+
+   rocoto:
+     tasks:
+       taskgroups: '{{ ["parm/wflow/prep.yaml", "parm/wflow/coldstart.yaml", "parm/wflow/post.yaml", "parm/wflow/verify.yaml"]|include }}'
+     metatask_vx_ens_member:
+       metatask_PointStat_mem#mem#:
+
+
+More information about configuring the ``rocoto:`` section can be found here.
```
Comment (MichaelLueken) [inline docs/UsersGuide/source/RunSRW.rst]: Should there be a link to the DefineWorkflow chapter here?

File: docs/UsersGuide/source/ConfigWorkflow.rst [hunk 2]
```diff
@@ -151,23 +151,23 @@ METplus Parameters
...
+   User-specified location of the top-level directory where NDAS prepbufr files used by METplus are located. This parameter needs to be set for both user-provided observations and for observations that are retrieved from the NOAA :term:`HPSS` (if the user has access) via the ``GET_OBS_NDAS`` task (activated in the workflow by automatically when using the taskgroup file ``parm/wflow/verify.yaml``). When pulling observations directly from NOAA HPSS, the data retrieved will be placed in this directory. Please note, this path must be defined as ``/<full-path-to-obs>/ndas/proc``. METplus is configured to verify near-surface variables hourly and upper-air variables at 00 and 12 UTC with NDAS prepbufr files.
```
Comment (MichaelLueken) [inline docs/UsersGuide/source/ConfigWorkflow.rst]: Should `by` be removed from line 170?

`task (activated in the workflow automatically when using the taskgroup file ``parm/wflow/verify.yaml``).
`
```suggestion
   User-specified location of the top-level directory where NDAS prepbufr files used by METplus are located. This parameter needs to be set for both user-provided observations and for observations that are retrieved from the NOAA :term:`HPSS` (if the user has access) via the ``GET_OBS_NDAS`` task (activated in the workflow automatically when using the taskgroup file ``parm/wflow/verify.yaml``). When pulling observations directly from NOAA HPSS, the data retrieved will be placed in this directory. Please note, this path must be defined as ``/<full-path-to-obs>/ndas/proc``. METplus is configured to verify near-surface variables hourly and upper-air variables at 00 and 12 UTC with NDAS prepbufr files. 
```

**Review by MichaelLueken — COMMENTED** [review]
@christinaholtNOAA Thank you very much for updating the documentation!  I have noted a minor issue in `ConfigWorkflow.rst` that should assist in readability and an issue in `RunSRW.rst` where it looks like you meant to add a link to the `DefineWorkflow` chapter, but this link wasn't provided.

**Review by MichaelLueken — APPROVED** [review]
@christinaholtNOAA Thank you very much for applying the changes!  Everything looks good.

File: docs/UsersGuide/source/ConfigWorkflow.rst [hunk 3]
```diff
@@ -692,7 +563,7 @@ Non-default parameters for the ``make_sfc_climo`` task are set in the ``task_mak
    Controls the size of the stack for threads created by the OpenMP implementation.
 
 ``SFC_CLIMO_DIR``: (Default: "")
-   The directory containing pre-generated surface climatology files to use when ``TN_MAKE_SFC_CLIMO`` is set to false.
+   The directory containing pre-generated surface climatology files to use when the ``MAKE_SFC_CLIMO`` is not meant to run.
```
Comment (gspetro-NOAA) [inline docs/UsersGuide/source/ConfigWorkflow.rst]: @christinaholtNOAA  Need the word "task" after "the ``MAKE_SFC_CLIMO``"

File: docs/UsersGuide/source/ConfigWorkflow.rst [hunk 4]
```diff
@@ -1155,7 +951,7 @@ These parameters are associated with the fixed (i.e., static) files. On `Level 1
    The location on disk of the static input files used by the ``make_orog`` task (i.e., ``orog.x`` and ``shave.x``). Can be the same as ``FIXgsm``.
 
 ``SFC_CLIMO_INPUT_DIR``: (Default: "")
-   The location on disk of the static surface climatology input fields, used by ``sfc_climo_gen``. These files are only used if ``RUN_TASK_MAKE_SFC_CLIMO: true``.
+   The location on disk of the static surface climatology input fields, used by ``sfc_climo_gen``. These files are only used if the ``MAKE_SFC_CLIMO`` is meant to run.
```
Comment (gspetro-NOAA) [inline docs/UsersGuide/source/ConfigWorkflow.rst]: Add "task" here, too.

**Comment by gspetro-NOAA** [comment]
@christinaholtNOAA When I build the docs, I get this warning:
`/Users/gillianpetro/holtSRW/docs/UsersGuide/source/Tutorial.rst:168: WARNING: undefined label: workflow-switches`
I think you'll just need to fix the crosslink on line 158 of Tutorial.rst to link to an appropriate section (based on the updated ConfigWorkflow.rst) or remove the reference if it doesn't make sense anymore.

**Comment by christinaholtNOAA** [comment]
@gspetro-NOAA I wasn't sure what to do there. I noticed the warning and left it because I do not feel comfortable updating the tutorial. My initial thought was that the warning kind of serves as a "to do" for that update. 

I can take the cross-reference out to fix the warning, but the text will still be quite wrong.

File: docs/UsersGuide/source/index.rst [hunk 5]
```diff
@@ -22,6 +22,7 @@ UFS Short-Range Weather App Users Guide
    InputOutputFiles
    LAMGrids
    ConfigWorkflow
+   DefineWorkflow
```
Comment (gspetro-NOAA) [inline docs/UsersGuide/source/index.rst]: @christinaholtNOAA It might make sense to add DefineWorkflow _after_ the RocotoInfo section.

**Comment by gspetro-NOAA** [comment]
@christinaholtNOAA I think that at a minimum, you should remove the line "For a detailed description of the workflow-switches: variables, see Section %s" because once that is committed to develop, the crosslink won't render properly in the docs (it will literally say "Section %s" instead of having a crosslink, which I think would be more confusing to users). You could also delete the workflow-switches section entirely and copy-paste your instructions from the RunSRW chapter on how to turn on plotting, but if that seems like a can of worms, don't worry about it. I can update the rest of that chapter when we do the next release.

**Comment by natalie-perlin** [comment]
Somehow, `workflow_switches` are no longer used in the current SRW version. I missed the moment they were taken out, and it's very unfortunate for a general user. What is the way for the user specify whether to run a particular task?..

**Comment by gspetro-NOAA** [comment]
@natalie-perlin Christina's edits to RunSRW.rst in this PR cover that. She wrote in this PR description that she will not update the full tutorial chapter at this time. I just need her to remove the one line in the Tutorial chapter that creates a broken link in order to approve. The rest will be updated later ahead of the release.

**Review by gspetro-NOAA — APPROVED** [review]
Thanks for the update! Code builds without error. :)

**Pull request merged**